test/common/grpc-service.js

```diff
@@ -1096,6 +1097,169 @@ describe('GrpcService', function() {
+      ProtoService.prototype.method = function() {
+        return duplexify.obj();
+      };
+      grpcService.getService_ = function(protoOpts) {
+        assert.strictEqual(protoOpts, PROTO_OPTS);
+        setImmediate(done);
+        return new ProtoService();
+      };
+
+      grpcService.requestWritableStream(PROTO_OPTS, REQ_OPTS);
+    });
+
+    it('should set the deadline', function(done) {
+      var createDeadline = GrpcService.createDeadline_;
+      var fakeDeadline = createDeadline(PROTO_OPTS.timeout);
+
+      GrpcService.createDeadline_ = function(timeout) {
+        assert.strictEqual(timeout, PROTO_OPTS.timeout);
+        return fakeDeadline;
+      };
+
+      ProtoService.prototype.method = function(reqOpts, grpcOpts) {
+        assert.strictEqual(grpcOpts.deadline, fakeDeadline);
+
+        GrpcService.createDeadline_ = createDeadline;
+        setImmediate(done);
+
+        return through.obj();
+      };
+
+      retryRequestOverride = function(_, retryOpts) {
+        return retryOpts.request();
+      };
+
+      grpcService.requestWritableStream(PROTO_OPTS, REQ_OPTS);
+    });
+
+    describe('getting gRPC credentials', function() {
+      beforeEach(function() {
+        delete grpcService.grpcCredentials;
+      });
+
+      describe('error', function() {
+        var error = new Error('err');
+
+        beforeEach(function() {
+          grpcService.getGrpcCredentials_ = function(callback) {
+            setImmediate(function() {
+              callback(error);
+            });
+          };
+        });
+
+        it('should execute callback with error', function(done) {
+          grpcService.requestWritableStream(PROTO_OPTS, REQ_OPTS)
+            .on('error', function(err) {
+              assert.strictEqual(err, error);
+              done();
+            });
+        });
+      });
+
+      describe('success', function() {
+        var authClient = {};
+
+        beforeEach(function() {
+          grpcService.getGrpcCredentials_ = function(callback) {
+            callback(null, authClient);
+          };
+        });
+
+        it('should make the gRPC request again', function(done) {
+          var stream = duplexify.obj();
+          ProtoService.prototype.method = function() {
+            return stream;
+          };
+          grpcService.getService_ = function() {
+            assert.strictEqual(grpcService.grpcCredentials, authClient);
+            return new ProtoService();
+          };
+
+          grpcService.requestWritableStream(PROTO_OPTS, REQ_OPTS)
+            .on('response', function(status) {
+              assert.equal(status, 'foo');
+              done();
+            })
+            .on('error', done);
+
+          setTimeout(function() {
+            stream.emit('status', 'foo');
+          }, 10);
+        });
+      });
+    });
+
+    describe('stream error', function() {
+      var authClient = {};
+
+      beforeEach(function() {
+        delete grpcService.grpcCredentials;
```

Comment:
Any reason we don't set `grpcCredentials` to an empty object or something similar here?